opentelemetry-exporter-cloud-trace/src/opentelemetry/tools/resource_detector.py

```diff
@@ -8,27 +10,71 @@
 _GCP_METADATA_URL_HEADER = {"Metadata-Flavor": "Google"}
 
 
-def get_gce_resources():
-    """ Resource finder for common GCE attributes
-
-        See: https://cloud.google.com/compute/docs/storing-retrieving-metadata
-    """
+def _get_google_metadata_and_common_attributes():
     token = attach(set_value("suppress_instrumentation", True))
     all_metadata = requests.get(
         _GCP_METADATA_URL, headers=_GCP_METADATA_URL_HEADER
     ).json()
     detach(token)
-    gce_resources = {
-        "host.id": all_metadata["instance"]["id"],
+    common_attributes = {
         "cloud.account.id": all_metadata["project"]["projectId"],
-        "cloud.zone": all_metadata["instance"]["zone"].split("/")[-1],
         "cloud.provider": "gcp",
-        "gcp.resource_type": "gce_instance",
+        "cloud.zone": all_metadata["instance"]["zone"].split("/")[-1],
     }
-    return gce_resources
+    return common_attributes, all_metadata
+
+
+def get_gce_resources():
+    """ Resource finder for common GCE attributes
+
+        See: https://cloud.google.com/compute/docs/storing-retrieving-metadata
+    """
+    (
+        common_attributes,
+        all_metadata,
+    ) = _get_google_metadata_and_common_attributes()
+    common_attributes.update(
+        {
+            "host.id": all_metadata["instance"]["id"],
+            "gcp.resource_type": "gce_instance",
+        }
+    )
+    return common_attributes
+
+
+def get_gke_resources():
+    """ Resource finder for GKE attributes
+
+    """
+    # The user must specify these environment variables via the Downward API
+    container_name = os.getenv("CONTAINER_NAME")
+    pod_namespace = os.getenv("NAMESPACE")
+    if not container_name or not pod_namespace:
```

Comment:
An empty string for CONTAINER_NAME, seems like the user added the field but not a value. I guess we should still attempt GKE detection instead of silently falling back to GCE. Changed!